src/main/java/com/vmware/vipclient/i18n/base/instances/TranslationMessage.java

```diff
@@ -96,7 +97,9 @@ public String getMessage(final Locale locale, final String component, final Stri
     		if (source != null && !source.isEmpty()) {
     			return FormatUtils.format(source, LocaleUtility.getSourceLocale(), args);
     		}
-    		return key;
+    		if (VIPCfg.getInstance().isProdMode())
```

Comment:
If-statement added for use case#2: A requested (target/default) message is not found; source message is also not found.